### DISCUSSION TITLE
dynamic fragments: including dynamically generated fragments in thymeleaf templates

### DISCUSSION THREAD

**Discussion by ag88** [thymeleaf/thymeleaf#941] [General]
It is possible to pass a fragment generated at run time to the thymeleaf template engine to be included in a template? 

For example, if my java codes generate a fragment for a menu to be included in the template. Is there a way to generate this dynamically and pass it to thymeleaf template engine to be `<div th:insert="<my generated fragment>" />` ?

**Comment by wimdeblauwe** [comment]
You can add the HTML as a model attribute:

```
@GetMapping
public String index(Model model) {
  String menuHtml = ... // generate html
  model.addAttribute("menuHtml", menuHtml);
}
```

Use `[(${menuHtml})]` to render it in your html:

```
<div>
[(${menuHtml})]
</div>
